hnn_core/params.py

```diff
@@ -24,18 +26,97 @@ def _count_evoked_inputs(d):
     return nprox, ndist
 
 
+def _read_json(fname):
+    """Read param values from a .json file.
+    Parameters
+    ----------
+    fname : str
+        Full path to the file (.json)
+
+    Returns
+    -------
+    params_input : dict
+        Dictionary of parameters
+    """
+    with open(fname) as json_data:
+        params_input = json.load(json_data)
+
+    return params_input
+
+def _read_legacy_params(fname):
+    """Read param values from a .param file (legacy).
+    Parameters
+    ----------
+    fname : str
+        Full path to the file (.param)
+
+    Returns
+    -------
+    params_input : dict
+        Dictionary of parameters
+    """
+
+    params_input = {}
+    with open(fname, 'r') as fp:
+        ln = fp.readlines()
+        for l in ln:
+            s = l.lstrip()
+            if s.startswith('#'):
+                continue
+            sp = s.split(':')
+            if len(sp) > 1:
+                key = sp[0].strip()
+                value = sp[1].strip()
+                if '.' in value or 'e' in value:
+                    try:
+                        params_input[key] = float(value)
+                    except ValueError:
+                        params_input[key] = value
+                else:
+                    try:
+                        params_input[key] = int(value)
+                    except ValueError:
+                        params_input[key] = value
+
+    return params_input
+
+
+def read_params(params_fname):
+    split_fname = op.splitext(params_fname)
+    try:
+        ext = split_fname[1]
+    except KeyError:
+        ext = ''
+
+    if ext == '.json': 
+        params_dict = _read_json(params_fname)
+    elif ext == '.param':
+        params_dict = _read_legacy_params(params_fname)
+    else:
+        raise ValueError('Unrecognized extension, expected one of' +
+                         ' .json, .param. Got %s' % ext)
+
+    if len(params_dict) == 0:
+        raise ValueError("Failed to read parameters from file: %s" %
+                         op.normpath(params_fname))
```

Comment:
check that the error is raised. Same with all the errors ...

Reply:
Check here:
https://github.com/hnnsolver/hnn-core/pull/80/commits/52130b1f676f1a86b5cbe43d6b90fccc2d126a38#diff-9ce7243ca4dfdfed86d90724fb56f15eR58